system/lib/wasmfs/wasmfs.cpp

```diff
@@ -81,7 +80,7 @@ __wasi_errno_t __wasi_fd_write(
     const uint8_t* buf = iovs[i].buf;
     __wasi_size_t len = iovs[i].buf_len;
 
-    file.write(buf, len);
+    static_cast<DataFile::Handle&>(file).write(buf, len);
```

Comment:
We need to actually check that we have a `DataFile` here. The user might be trying to write to a directory, for example.